scripts/spi_flash/spi_flash.py

```diff
@@ -1397,9 +1398,10 @@ def sdcard_upload(self):
         output("Validating Upload...")
         try:
             finfo = self.fatfs.get_file_info(fw_path)
-            with self.fatfs.open_file(fw_path, 'r') as sd_f:
+            with self.fatfs.open_file(fw_path, 'rb') as sd_f:
```

Comment:
FYI, "b" is not a supported mode.  The SDCardFile class will remove it, so this change has no effect.